pkg/reconciler/openshift/tektonaddon/tektonaddon.go

```diff
@@ -369,40 +358,36 @@ func (r *Reconciler) ensureClusterTasks(ctx context.Context, ta *v1alpha1.Tekton
 // and if installer set which already exist is of older version then it deletes and return false to create a new
 // installer set
 func checkIfInstallerSetExist(ctx context.Context, oc clientset.Interface, relVersion string,
-	ta *v1alpha1.TektonAddon, component string) (bool, error) {
+	labelSelector string) (bool, error) {
 
-	// Check if installer set is already created
-	compInstallerSet, ok := ta.Status.AddonsInstallerSet[component]
-	if !ok {
-		return false, nil
+	installerSets, err := oc.OperatorV1alpha1().TektonInstallerSets().
+		List(ctx, metav1.ListOptions{
+			LabelSelector: labelSelector,
+		})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
 	}
 
-	if compInstallerSet != "" {
+	if len(installerSets.Items) == 1 {
 		// if already created then check which version it is
-		ctIs, err := oc.OperatorV1alpha1().TektonInstallerSets().
-			Get(ctx, compInstallerSet, metav1.GetOptions{})
-		if err != nil {
-			if errors.IsNotFound(err) {
-				return false, nil
-			}
-			return false, err
-		}
-
-		version, ok := ctIs.Annotations[tektoninstallerset.ReleaseVersionKey]
+		version, ok := installerSets.Items[0].Labels[tektoninstallerset.ReleaseVersionKey]
 		if ok && version == relVersion {
 			// if installer set already exist and release version is same
 			// then ignore and move on
 			return true, nil
 		}
+	}
 
-		// release version doesn't exist or is different from expected
-		// deleted existing InstallerSet and create a new one
-
-		err = oc.OperatorV1alpha1().TektonInstallerSets().
-			Delete(ctx, compInstallerSet, metav1.DeleteOptions{})
-		if err != nil {
-			return false, err
-		}
+	// release version doesn't exist or is different from expected
+	// deleted existing InstallerSet and create a new one
+	if err = oc.OperatorV1alpha1().TektonInstallerSets().
+		DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: labelSelector,
+		}); err != nil {
+		return false, err
```

Comment:
this should work but ideally, we should loop through the list and call individual delete() after ensuring `DeletionTimestamp` is not set in metadata of each installerset. 

 We could be here even after a previous reconcile loop has called `DeleteCollection` as the installerset might take few seconds to clean up (finalize) what it has created.

Reply:
ie, the DeletionCollection() is just an API call, so we cannot gurantee when the installerSet will be completely removed.

Mentioning this here, to make sure that we are all on the same page.
cc @sm43